tempodb/tempodb.go

```diff
@@ -383,6 +383,13 @@ func (rw *readerWriter) pollBlocklist() {
 			metricBlocklistErrors.WithLabelValues(tenantID).Inc()
 			level.Error(rw.logger).Log("msg", "error polling blocklist", "tenantID", tenantID, "err", err)
 		}
+		if len(blockIDs) == 0 {
```

Comment:
oh, yikes, I missed this but this is in the wrong spot.  This should be down around line 434.  If, at that point, the len(blocklist) == 0 then we should delete from the map.

Also, the compactedBlocklist and normal blocklist need to be handled independently.  It's possible a tenant will have no normal blocks, but still have compacted blocks.

Reply:
Ok after further review and an offline discussion with @dgzlopes it appears this functionality actually needs to be line ~379.

In an object storage system like s3 or gcs there are no "folders" per se, just objects with a key path.  In this case if there are no objects there will be no tenant.  So, after retrieving the tenants on line 374 we need to check and remove every tenant in the blocklist maps that are not in the tenant list.